### PR TITLE
Implementation of 'minirand' Dynamic Randomization Algorithm

### PR DESCRIPTION
**Pull request by kamilsi**
### Overview
This PR addresses the implementation of dynamic randomization, focusing on the integration of the 'minirand' algorithm (DOI: 10.1080/03610918.2019.1619765.).

### Issue Addressed
Implement Dynamic Randomization Using 'minirand' Algorithm #12 (partially)

### Key Achievements

1. Algorithm Adaptation: Adapted Michał Seweryn's `varrand` function for the 'minirand' algorithm, aligning it with our coding standards in the unbiased package.
2. Documentation: Updated internal documentation to cover the newly integrated varrand (now `randomize_dynamic()`) function, providing clear usage guidelines and parameter explanations.
3. Testing: Developed a comprehensive suite of unit tests to validate the functionality and reliability of the function, ensuring compliance with our testing standards.

### Pending Tasks
The following tasks remain to be addressed in subsequent phases of this project:

1. PR Review and Merge: Solicit feedback on the current integration and proceed with merging this PR after thorough review.
1. Further Development in Branches:
    1. Vignette Development Branch: Initiate a branch dedicated to creating a vignette. This vignette will employ Monte Carlo simulations to demonstrate the efficacy of the 'minirand' method, especially in complex scenarios with multiple covariates and a limited number of subjects. The aim is to showcase its superiority in achieving balance compared to traditional random or block randomization methods.
    2. API Integration Branch: Start a separate branch focused on incorporating the dynamic randomization process into our existing API. This will involve endpoint creation, adhering to our architectural standards and ensuring seamless integration of the 'minirand' algorithm within our system.

### CODE REVIEW

**Review by salatak — COMMENTED** [review]
Code checked. Aesthetic. You could try to optimize it even more. My comments only concern the description of the functions.

File: DESCRIPTION [hunk 1]
```diff
@@ -9,14 +9,18 @@ License: MIT + file LICENSE
 Imports: 
     checkmate,
     dbplyr,
-    plumber
+    plumber,
+    mathjaxr,
```
[the remark this answers is not in the document]
Comment (kamilsi) [reply DESCRIPTION:13]: We still have things like 'patient allocation ratios (\(a_{k}\))' in description

File: R/randomize-dynamic.R [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+#'        Function used to compute within-arm variability, must be one of:
+#'        `sd`, `var`, `range`, defaults to `var`
+#' @param p `numeric()`\cr
+#'        single value, proportion of randomness (0, 1) in the randomization
+#'        vs determinism, defaults to 85% deterministic
+#'
+#' @return `character()`\cr
+#'         name of the arm assigned to the patient
+#' @examples
+#' n_at_the_moment <- 10
```
[the remark this answers is not in the document]
Comment (kamilsi) [reply R/randomize-dynamic.R:83]: Good idea, added.

File: R/randomize-dynamic.R [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+#'        last row contains the new patient with empty string for `arm`
+#' @param weights `numeric()`\cr
+#'        vector of positive weights, equal in length to number of covariates,
+#'        numbered after covariates, defaults to equal weights
+#' @param method `character()`\cr
+#'        Function used to compute within-arm variability, must be one of:
+#'        `sd`, `var`, `range`, defaults to `var`
+#' @param p `numeric()`\cr
+#'        single value, proportion of randomness (0, 1) in the randomization
+#'        vs determinism, defaults to 85% deterministic
```
[the remark this answers is not in the document]
Comment (kamilsi) [reply R/randomize-dynamic.R:78]: The descriptions for arms and ratio are indeed included in our documentation, but they are not directly under the randomize_minimisation_pocock function. Instead, these details are inherited from the documentation of the randomize_simple function through the use of the @inheritParams randomize_simple tag in our R documentation.

This means that the arms and ratio parameters are defined and described in the randomize_simple function, and by using @inheritParams, we avoid duplicating the same information in the randomize_minimisation_pocock documentation. This approach keeps our documentation DRY (Don't Repeat Yourself) and ensures consistency across related functions.

File: R/randomize-simple.R [hunk 4]
```diff
@@ -5,9 +5,10 @@
...
-#'  Arm names.
-#' @param ratio `numeric()`\cr
-#'  Ratio of patient assignment to each arm. Must be the same length as `arms`.
+#'        Arm names.
+#' @param ratio `integer()`\cr
+#'        Vector of positive integers (0 is allowed), equal in length to number
+#'        of arms, named after arms, defaults to equal weight
 #'
 #' @return Selected arm assignment.
 #'
```
[the remark this answers is not in the document]
Comment (kamilsi) [reply R/randomize-simple.R:14]: Thank you, I fixed that by relaxing the check to "integerish" values. Now you can skip "L" suffix.

**Pull request merged**